client/packages/common/src/hooks/useNativeClient/useNativeClient.ts

```diff
@@ -0,0 +1,157 @@
+import { uniqWith } from 'lodash';
```

Comment:
File renamed since it will be common for Android/Electron.

Apart from file change:
* Use polling instead of callback to client to check discovered servers (see PR description)
* Autoconnect is not done here (to consolidate that logic for both clients)
* goBackToDiscovery is added (to go back to discovery from initialisation/login
* Server is match by hardwareId and port